src/lib/Occurrence.hs

```diff
@@ -84,6 +87,14 @@ class MaxPlus a where
   max :: a -> a -> a
   plus :: a -> a -> a
 
+instance (MaxPlus a) => MaxPlus (M.IntMap a) where
```

Comment:
Well `IntMap` is a bit of an implementation detail of `NameMap`. Perhaps implement this for `NameMap` (since we already have `unionWith`) and then derive for `NameMapE`?

Reply:
Done.